### PR TITLE
Crew Cache Cleanup

### PR DESCRIPTION
**Pull request by satmandu**
- Adds trailing slash to CREW_CACHE_DIR using technique from https://stackoverflow.com/a/12393692
- Link cached archive to and from CREW_BREW_DIR using hard links if possible, which is much faster than copying.
- Don't copy archive to cache if it is from cache

Works properly:
- [x] x86_64

### CODE REVIEW

File: bin/crew [hunk 1]
```diff
@@ -578,36 +578,58 @@ def download
...
+          cachefile = ''
+        else
+          puts "Archive file exists in cache".lightgreen if @opt_verbose
           if Digest::SHA256.hexdigest( File.read(cachefile) ) == sha256sum then
-            FileUtils.cp cachefile, CREW_BREW_DIR, verbose: @fileutils_verbose
-            puts "Archive found in cache".lightgreen
-            return {source: source, filename: filename}
+            begin
+              # Hard link cached file if possible.
+              FileUtils.ln cachefile, CREW_BREW_DIR, verbose: @fileutils_verbose
```
Comment (uberhacker) [inline bin/crew:593]: Why a hard link and why call it cachefile when you are linking to a directory?

File: lib/const.rb [hunk 2]
```diff
@@ -35,11 +35,13 @@
   HOME = CREW_PREFIX + ENV['HOME']
 end
 
+# File.join ensures a trailing slash if one does not exist.
 if ENV['CREW_CACHE_DIR'].to_s == ''
```
Comment (uberhacker) [inline lib/const.rb:39]: `if ENV['CREW_CACHE_DIR'].to_s.empty`?

File: bin/crew (hunk 1 above)
Comment (satmandu) [reply bin/crew:593]: CREW_BREW_DIR is a dir, but `ln filename dir` should make a hard linked file in `dir`, no?

File: lib/const.rb (hunk 2 above)
Comment (satmandu) [reply lib/const.rb:39]: Sure, but that was the existing code...

File: bin/crew (hunk 1 above)
Comment (uberhacker) [reply bin/crew:593]: I suppose that is true if the directory exists.

File: lib/const.rb (hunk 2 above)
Comment (satmandu) [reply lib/const.rb]: I changed all  `.to_s == ''` to `.to_s.empty` .

**Pull request merged**